bazel/grpc_build_system.bzl

```diff
@@ -259,7 +262,74 @@ def ios_cc_test(
             deps = ios_test_deps,
         )
 
-def grpc_cc_test(name, srcs = [], deps = [], external_deps = [], args = [], data = [], uses_polling = True, language = "C++", size = "medium", timeout = None, tags = [], exec_compatible_with = [], exec_properties = {}, shard_count = None, flaky = None, copts = [], linkstatic = None):
+def expand_tests_for_each_poller_and_engine(name, srcs, deps, tags, args, exclude_pollers, uses_event_engine):
+    """Common logic used to parameterize tests for every poller and EventEngine.
+
+    Args:
+        name: base name of the test
+        srcs: source files
+        deps: base deps
+        tags: base tags
+        args: base args
+        exclude_pollers: list of poller names to exclude for this set of tests.
+        uses_event_engine: set to False if the test is not sensitive to
+            EventEngine implementation differences
+
+    Returns:
+        A list of dictionaries containing modified values of name, srcs, deps, tags, and args.
+    """
+    poller_config = []
+
+    # On linux we run the same test with the default EventEngine, once for each
+    # poller
+    for poller in POLLERS:
+        if poller in exclude_pollers:
+            continue
+        poller_config.append({
+            "name": name + "@poller=" + poller,
+            "srcs": srcs,
+            "deps": deps,
+            "tags": (tags + EVENT_ENGINES["default"]["tags"] + [
+                "no_windows",
+                "no_mac",
+                "bazel_only",
+            ]),
+            "args": args + ["--poller=" + poller],
+        })
+
+    # Now generate one test for each subsequent EventEngine, all using the
+    # default poller.
+    if not uses_event_engine:
+        # The poller tests exercise the default engine on Linux. This test
+        # handles other platforms.
+        poller_config.append({
+            "name": name,
+            "srcs": srcs,
+            "deps": deps,
+            "tags": tags + ["no_linux"],
+            "args": args,
+        })
+    else:
+        for engine_name, engine in EVENT_ENGINES.items():
```

Comment:
I looked at the list of bazel tests that ran on this PR and I never saw the "@engine=" in the name of a test, which seems suspicious?

Reply:
Done.